Common/Data/Market/QuoteBar.cs

```diff
@@ -0,0 +1,495 @@
+        //public decimal Price;
+
+        //Symbol of Asset.
+        //In Base Class: public Symbol Symbol;
+
+        //In Base Class: DateTime Of this QuoteBar
+        //public DateTime Time;
+
+        /// <summary>
+        /// Default initializer to setup an empty quotebar.
+        /// </summary>
+        public QuoteBar()
+        {
+            Symbol = Symbol.Empty;
+            Time = new DateTime();
+            Bid = new Bar();
+            Ask = new Bar();
+            AvgBidSize = 0;
+            AvgAskSize = 0;
+            Open = 0;
+            High = 0;
+            Low = 0;
+            Close = 0;
+            Value = 0;
+            Period = TimeSpan.FromMinutes(1);
+            DataType = MarketDataType.QuoteBar;
+        }
+
+        /// <summary>
+        /// Cloner constructor for implementing fill forward. 
+        /// Return a new instance with the same values as this original.
+        /// </summary>
+        /// <param name="original">Original quotebar object we seek to clone</param>
+        public QuoteBar(QuoteBar original)
+        {
+            Symbol = original.Symbol;
+            Time = new DateTime(original.Time.Ticks);
+            Bid = original.Bid;
+            Ask = original.Ask;
+            AvgBidSize = original.AvgBidSize;
+            AvgAskSize = original.AvgAskSize;
+            Open = original.Open;
+            High = original.High;
+            Low = original.Low;
+            Close = original.Close;
+            Value = original.Close;
+            Period = original.Period;
+            DataType = MarketDataType.QuoteBar;
+        }
+
+        /// <summary>
+        /// Initialize Quote Bar with Bid(OHLC) and Ask(OHLC) Values:
+        /// </summary>
+        /// <param name="time">DateTime Timestamp of the bar</param>
+        /// <param name="symbol">Market MarketType Symbol</param>
+        /// <param name="bidopen">Decimal Opening Price of Bid</param>
+        /// <param name="bidhigh">Decimal High Price of this bid bar</param>
+        /// <param name="bidlow">Decimal Low Price of this bid bar</param>
+        /// <param name="bidclose">Decimal Close price of this bid bar</param>
+        /// <param name="avgbidsize">Average bid size over period</param>
+        /// <param name="askopen">Decimal Opening Price of Ask</param>
+        /// <param name="askhigh">Decimal High Price of this ask bar</param>
+        /// <param name="asklow">Decimal Low Price of this ask bar</param>
+        /// <param name="askclose">Decimal Close price of this ask bar</param>
+        /// <param name="avgasksize">Average ask size over period</param>
+        /// <param name="period">The period of this bar, specify null for default of 1 minute</param>
+        public QuoteBar(DateTime time, Symbol symbol, decimal bidopen, decimal bidhigh, decimal bidlow, decimal bidclose, long avgbidsize, decimal askopen, decimal askhigh, decimal asklow, decimal askclose, long avgasksize, TimeSpan? period = null)
```

Comment:
Well, yes. 
If I read an csv file with quote bars, I can make Bid/Ask `<IBar>` before I call the constructor...